pandas/core/indexes/base.py

```diff
@@ -206,10 +206,10 @@ class Index(IndexOpsMixin, PandasObject):
 
     # tolist is not actually deprecated, just suppressed in the __dir__
     _deprecations = (
-        IndexOpsMixin._deprecations
-        | DirNamesMixin._deprecations
-        | frozenset(["tolist", "contains", "dtype_str", "get_values", "set_value"])
-    )
+        PandasObject._deprecations
+        | IndexOpsMixin._deprecations
+        | frozenset(["asobject", "contains", "dtype_str", "get_values", "set_value"])
+    )  # type: FrozenSet[str]
```

Comment:
Same comment here

Reply:
I added the type hints after complaints from mypy. I agree it's strange it doesn't infer and I'll see if something can be done to make it infer.

Reply:
I think should be able to infer as long as you construct with values. Empty construction complains because it is not known then what types of value the object should hold. 

Not sure if FrozenSet would act differently but that’s how builtins work

Reply:
Same here. A frozenset that only contains strings, could conceivably have non-strings, and I think that's what mypy complains about.